splunk-devops/src/main/java/com/splunk/splunkjenkins/JdkSplunkLogHandler.java

```diff
@@ -77,6 +85,28 @@ public void flush() {
         SplunkLogService.getInstance().sendBatch(copyList, EventType.LOG);
     }
 
+    @Override
+    public boolean isLoggable(LogRecord record) {
+        if (!record.getLoggerName().contains(JdkSplunkLogHandler.class.getName())
+                && !record.getLoggerName().contains(CustomLoggersConfig.class.getName())) { // ignores self references
+            CustomLoggersConfig clConfig = CustomLoggersConfig.get();
+            if (clConfig != null) {
+                for (CustomLoggerItem clItem : clConfig.getCustomLoggers()) {
```

Comment:
Existing logic https://github.com/jenkinsci/splunk-devops-plugin/blob/master/splunk-devops/src/main/java/com/splunk/splunkjenkins/LoggingInitStep.java#L40
`Logger.getLogger(rootLoggerName).addHandler(JdkSplunkLogHandler.LogHolder.LOG_HANDLER);` registers to root logger, that means every log record will loop the config items. Assume you have 10 targets configured, and 1000 log records per second, that means 10000 additional checks even if the logger is not in the config list